screens/AllTasks.tsx

```diff
@@ -19,16 +18,10 @@ export default function AllTasks() {
         "SELECT category FROM Tasks GROUP BY category"
       );
       setCategs(allCategs);
-
-      // for(const cat of allCategs){
-      //   const categTasks =await db.getAllAsync<Task>("SELECT * FROM Tasks WHERE category = ? ",[cat.category]);
-
-      //   setCategs((curState)=>{return(     )    })
-      // }
     }
 
     fetchCats();
```

Comment:
Although it might work now. It's not ideal way to set set inside a asyn function like this here in this case.

Say the component renders before you get your response from the database then `setCategs` will no longer be valid, since react will re-create all the states and functions.

The right way to do this would be return `allCategs` inside fetchCats and set in the `then`, eg. `fetchCats().then(() => { setCategs() })`.

Reply:
Also add the fetchCats to your `useEffect` dependencies.